vendor-extra/ContentPageBundle/src/Controller/ContentController.php

```diff
@@ -4,24 +4,42 @@
 
 namespace Libero\ContentPageBundle\Controller;
 
+use DOMNodeList;
 use FluentDOM;
+use FluentDOM\DOM\Element;
 use GuzzleHttp\ClientInterface;
+use Libero\Views\ViewConverter;
 use Psr\Http\Message\ResponseInterface;
+use Punic\Misc;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Twig\Environment;
 use UnexpectedValueException;
+use function array_merge;
 
 final class ContentController
 {
     private $client;
+    private $converter;
     private $service;
+    private $template;
+    private $twig;
 
-    public function __construct(ClientInterface $client, string $service)
-    {
+    public function __construct(
+        ClientInterface $client,
+        string $service,
+        Environment $twig,
+        ViewConverter $converter,
+        string $template
+    ) {
         $this->client = $client;
         $this->service = $service;
+        $this->twig = $twig;
+        $this->converter = $converter;
```

Comment:
the number of dependencies here suggests to introduce some collaborator object to hide them, still unclear how to abstract it